tests/functional/modules/test_zos_volume_init_func.py

```diff
@@ -0,0 +1,376 @@
+        verify_volid=params['verify_volid']
+        )
+
+    for result in results.contacted.values():
+        # assert result.get('changed') is True
+        assert result['rc'] == 0
+
+    # bring volume back online
+    hosts.all.zos_operator(cmd=f"vary {TEST_VOL_ADDR},online")
+
+
+@pytest.mark.parametrize(
+    "params", [
+        # min params test with index : true
+        ({
+            'address': TEST_VOL_ADDR,
+            'verify_offline': False,
+            'volid': TEST_VOL_SER,
+            'index' : True
+        }),
+        # min params test with index : false
+        ({
+            'address': TEST_VOL_ADDR,
+            'verify_offline': False,
+            'volid': TEST_VOL_SER,
+            'index' : False,
+            'sms_managed' : False # default is True, which cannot be with no index.
+        }),
+    ]
+)
+def test_index_param(ansible_zos_module, params):
+    hosts = ansible_zos_module
+
+    # take volume offline
+    hosts.all.zos_operator(cmd=f"vary {TEST_VOL_ADDR},offline")
+
+    results = hosts.all.zos_volume_init(**params)
+
+    for result in results.contacted.values():
+        assert result.get("changed") is True
+        assert result.get('rc') == 0
+        content_str = ''.join(result.get("content"))
+        if params['index']:
+            assert INDEX_CREATION_SUCCESS_MSG in content_str
+        else:
+            assert INDEX_CREATION_SUCCESS_MSG not in content_str
+
+    # bring volume back online
+    hosts.all.zos_operator(cmd=f"vary {TEST_VOL_ADDR},online")
+
+
+# check that correct volume_addr is assigned to correct volid
+def test_volid_address_assigned_correctly(ansible_zos_module):
+    hosts = ansible_zos_module
+
+    params = {
+        'address': TEST_VOL_ADDR,
+        'verify_offline': False,
+        'volid': TEST_VOL_SER,
+    }
+    # take volume offline
+    hosts.all.zos_operator(cmd=f"vary {TEST_VOL_ADDR},offline")
+
+    results = hosts.all.zos_volume_init(**params)
+
+    # bring volume back online
+    hosts.all.zos_operator(cmd=f"vary {TEST_VOL_ADDR},online")
+    # display command to print device status, volser and addr should correspond
+
+    display_cmd_output = list(hosts.all.zos_operator(cmd=f"D U,VOL={TEST_VOL_SER}").contacted.values())[0].get('content')
+    display_cmd_output = ''.join(s for s in display_cmd_output)
+
+    for result in results.contacted.values():
+        assert result.get("changed") is True
+        assert result.get('rc') == 0
+
+        # The display command issued queries a volume called $TEST_VOL_SER. The
+        # expected return values are 'IEE455I UNIT STATUS NO DEVICES WITH
+        # REQUESTED ATTRIBUTES' or a line with several attributes including unit
+        # address (expected value $TEST_VOL_ADDR) and volume serial (expected
+        # value $TEST_VOL_SER). If those two match, then the 'volid' parameter
+        # is correctly assigned to the 'address' parameter.
+        assert TEST_VOL_SER in display_cmd_output
```

Comment:
This assertion concerns me: it is running a test on the same data in each loop, since it is assigned outside of the loop, and no indexing is involved.  I think this means 1 successful match suffices for all results.contacted, which I don't think is the goal... shouldn't it check each result contacted separately, or should it only be called once outside the loop?